onelogin_aws_cli/__init__.py

```diff
@@ -75,6 +79,26 @@ def get_saml_assertion(self):
 
         self.saml = saml_resp
 
+    def get_ip_address(self):
+        # if ip address has been hard coded in config file, use that
+        ip_address = self.config.get('ip_address')
+        if ip_address is not None:
+            return ip_address
+
+        # otherwise get ip address dynamically from api
+        uri = self.config.get('ip_address_api_uri')
+
+        # if option has not been configured
+        if uri is None:
+            return None
+
+        # fetch ip from the uri
+        request = urllib.request.Request(uri)
```

Comment:
I'm also inclined to think a command line param would make this an easier solution.  Then people wanting to implement auto-determination could do

```
onelogin-aws-login --ip $(curl https://api.ipify.org)
```

Reply:
At my company, we've configured the mutli account app so that our users only have to run `onelogin-aws-login` with no arguments to get access to their roles, so I could see how add this to the cli for every call would get annoying. 

I am torn between:
 - "provide user with method to determine IP in any way they choose as different networks could present many different ways to get a public IP accurately"
and
 - "make the app behaviour able to be completely defined all configurable through the config file for enterprise/wide roll outs"

I was thinking `auto_determine_ip` for run of the mill "we're behind a normal adsl/cable/fibre router/whatever with a nat" and allowing specification of `ONELOGIN_AWS_CLI_IP=$(my_complex_ip_determination.sh)` placed on a `~/.bash_profile` or something...

Neither of my suggestions are amazing solutions, but this feels slippery just allowing either specific ip or call an http endpoint.

Reply:
Also... I didn't realise you could do this user policies in Onelogin. Thank you @thnee ! 😄

Reply:
I totally appreciate that you a) don't want to tie the implementation to a specific ip lookup service, and b) offering a CLI option is a great addition that works for some people. However, in our use case it is important to not force the user to have to specify a bunch of CLI flags every time. And setting it via an environment variable would mean that if the IP changes while the user is logged in, it would not be picked up. So it seems that I would prefer to have some possibility to  somehow configure an API URI in onelogin-aws-cli. Do you think there is some way we can make that work? I definitely don't mind making the code more generic, but what aspects would we have to support in that case?

Reply:
How about this: What if I add `ip_address_lookup_command` as a config file option, which is a shell command that gets executed by the `subprocess` module? That command can then be configured to a value like `curl https://api.ipify.org` by the user.